hazelcast/src/test/java/com/hazelcast/query/impl/IndexSplitBrainTest.java

```diff
@@ -30,45 +27,65 @@
 
 @RunWith(HazelcastSerialClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
-public class IndexSplitBrainTest extends HazelcastTestSupport {
+public class IndexSplitBrainTest extends SplitBrainTestSupport {
 
-    private TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
-    
-    @Test
-    public void testIndexDoesNotReturnStaleResultsAfterSplit() {
-        String mapName = randomMapName();
-        Config config = newConfig(mapName);
+    private String mapName;
+    private String key;
+    private ValueObject value;
 
-        HazelcastInstance h1 = factory.newHazelcastInstance(config);
-        HazelcastInstance h2 = factory.newHazelcastInstance(config);
+    @Override
+    protected void onBeforeSetup() {
+        mapName = randomMapName();
+    }
 
-        warmUpPartitions(h1, h2);
-        String key = generateKeyOwnedBy(h1);
+    @Override
+    protected int[] brains() {
+        return new int[] {1,1};
+    }
 
-        final ValueObject value = new ValueObject(key);
+    @Override
+    protected void onBeforeSplitBrainCreated(HazelcastInstance[] instances)
+            throws Exception {
+        warmUpPartitions(instances);
+        key = generateKeyOwnedBy(instances[0]);
+        value = new ValueObject(key);
 
-        final IMap<String, ValueObject> map1 = h1.getMap(mapName);
-        final IMap<String, ValueObject> map2 = h2.getMap(mapName);
+        final IMap<String, ValueObject> map1 = instances[0].getMap(mapName);
+        final IMap<String, ValueObject> map2 = instances[1].getMap(mapName);
 
         map1.put(key, value);
-        assertNotNull("Entry should exist in map2 before split", map2.get(key));
-
-        // create split 
-        closeConnectionBetween(h1, h2);
-        assertClusterSizeEventually(1, h1);
-        assertClusterSizeEventually(1, h2);
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run()
+                    throws Exception {
+                assertNotNull("Entry should exist in map2 before split", map2.get(key));
```

Comment:
👍 when [this assertion failed](https://github.com/hazelcast/hazelcast/issues/9037#issue-180588154) there was no initial partition warmup in test code, so maybe the failure was due to initial partition assignments not being completed on the second instance in the time between `get` and `put`